scrabbler/strategy.py

```diff
@@ -1,17 +1,29 @@
+import random
+
 class Strategy:
     '''
     Class representing a player's Strategy. The three algorithms should be a 
     subclass of this class and override the method choose move.
     '''
-    def choose_move(self, game, rack):
+    def choose_move(self, game, rack, score_diff, opponents_rack, dictionary):
         raise Exception('Method not defined!')
 
 class BaselineStrategy(Strategy):
-    def choose_move(self, game, rack):
-        """ method that chooses the best move"""
+    def choose_move(self, game, rack, score_diff, opponents_rack, dictionary):
+        """ method that chooses a move based on this strategy"""
 
         # return the valid move with the highest score
         valid_moves = game.find_valid_moves(rack)
         if len(valid_moves) == 0:
             return None
         return valid_moves[0]
+
+class RandomStrategy(Strategy):
+    """ Strategy Class that chooses a random move instead of the move resulting in the highest score """
+    def choose_move(self, game, rack, score_diff, opponents_rack, dictionary):
```

Comment:
If you need extra parameters in your `choose_move` method, I don't think you need to also add them to the `Strategy` class or the `BaselineStrategy` class. You can just add the parameters to your `choose_move` method, but make them optional, so that the `choose_move` method in your class keeps the same behavior as its parent class.